src/lib.rs

```diff
@@ -112,11 +117,17 @@ pub fn compile_book(book: &mut Book, opts: CompileOpts) -> Result<CompileResult,
   Ok(CompileResult { core_book, labels, warns })
 }
 
-pub fn desugar_book(book: &mut Book, opts: CompileOpts) -> Result<Vec<Warning>, Info> {
+pub fn desugar_book(
+  book: &mut Book,
+  args: Option<Vec<Term>>,
```

Comment:
Same here, desugar shouldn't receive arguments but instead just desugar main to a match or something